chartlets.js/packages/lib/src/actions/handleHostStoreChange.ts

```diff
@@ -43,29 +45,81 @@ export function handleHostStoreChange() {
-    invokeCallbacks(callbackRequests);
+
+  const filteredCallbackRequests = callbackRequests.filter(
+    (callbackRequest): callbackRequest is CallbackRequest =>
+      callbackRequest !== undefined,
+  );
+  if (filteredCallbackRequests && filteredCallbackRequests.length > 0) {
+    invokeCallbacks(filteredCallbackRequests);
   }
 }
 
-function getCallbackRequests(
+// Exporting for testing only
+export function getCallbackRequests(
   propertyRefs: PropertyRef[],
   contributionsRecord: Record<string, ContributionState[]>,
   hostStore: HostStore,
-): CallbackRequest[] {
-  return propertyRefs.map((propertyRef) => {
-    const contributions = contributionsRecord[propertyRef.contribPoint];
-    const contribution = contributions[propertyRef.contribIndex];
-    const callback = contribution.callbacks![propertyRef.callbackIndex];
-    const inputValues = getInputValues(
-      callback.inputs!,
-      contribution,
+): (CallbackRequest | undefined)[] {
+  const { lastCallbackInputValues } = store.getState();
+  return propertyRefs.map((propertyRef) =>
+    getCallbackRequest(
+      propertyRef,
+      lastCallbackInputValues,
+      contributionsRecord,
       hostStore,
-    );
-    return { ...propertyRef, inputValues };
-  });
+    ),
+  );
 }
 
+// This is a dummy function created to memoize the _inputValues values from
+// getCallbackRequest()
+const _getInputValues = (_inputValues: unknown[]): unknown[] => {
+  return _inputValues;
+};
+const memoizedInputValues = memoize(_getInputValues);
+
+const getCallbackRequest = (
+  propertyRef: PropertyRef,
+  lastCallbackInputValues: Record<string, unknown[]> | undefined,
+  contributionsRecord: Record<string, ContributionState[]>,
+  hostStore: HostStore,
+) => {
+  const contribPoint: string = propertyRef.contribPoint;
+  const contribIndex: number = propertyRef.contribIndex;
+  const callbackIndex: number = propertyRef.callbackIndex;
+  const contributions = contributionsRecord[contribPoint];
+  const contribution = contributions[contribIndex];
+  const callback = contribution.callbacks![callbackIndex];
+  const _inputValues = getInputValues(
+    callback.inputs!,
+    contribution,
+    hostStore,
+  );
+
+  const inputValues = memoizedInputValues(_inputValues);
+
+  const callbackId = `${contribPoint}-${contribIndex}-${callbackIndex}`;
+  if (lastCallbackInputValues) {
+    const lastInputValues = lastCallbackInputValues[callbackId];
+    if (lastInputValues && shallowEqualArrays(lastInputValues, inputValues)) {
+      // We no longer log, as the situation is quite common
+      // Enable error logging for debugging only:
+      // console.groupCollapsed("Skipping callback request");
+      // console.debug("inputValues", inputValues);
+      // console.groupEnd();
+      return undefined;
+    }
+    lastCallbackInputValues[callbackId] = inputValues;
+    store.setState({
+      lastCallbackInputValues: { ...lastCallbackInputValues },
+    });
```

Comment:
This is more comprehensive in my eyes:

```suggestion
    store.setState({
      lastCallbackInputValues: { 
        ...lastCallbackInputValues,  
        [callbackId]: inputValues,
      },
    });
```